sfdoc/publish/tasks.py

```diff
@@ -10,294 +10,368 @@
+    zip_file = BytesIO(response.content)
+    with TemporaryDirectory(f"bundle_{bundle.pk}_") as tempdir:
+        utils.unzip(zip_file, tempdir, recursive=True, ignore_patterns=["*/assets/*"])
+
+        rootpath = utils.find_bundle_root_directory(tempdir)
+        assert os.path.exists(os.path.join(rootpath, "log.txt"))
+        utils.sync_directories(rootpath, path)
+        assert os.path.exists(os.path.join(path, "log.txt"))
+
 
+def _download_repository_state_from_s3(path):
+    assert os.path.exists(path)
+    logger.info(
+        "Downloading repository state from S3 %s", AWS_S3_PUBLISHED_HTML_REPOSITORY_URL
+    )
 
-def _process_bundle(bundle, path, enforce_no_duplicates = True):
+    # update our clean local copy
+    utils.s3_sync(AWS_S3_PUBLISHED_HTML_REPOSITORY_URL, settings.LOCAL_REPOSITORY_CACHE)
+    assert os.path.exists(settings.LOCAL_REPOSITORY_CACHE)
+
+    if len(os.listdir(settings.LOCAL_REPOSITORY_CACHE)):
+        assert os.path.exists(os.path.join(settings.LOCAL_REPOSITORY_CACHE, "log.txt"))
+
+    # copy it to the temporary working directory for this bundle
+    utils.sync_directories(settings.LOCAL_REPOSITORY_CACHE, path)
+    assert os.path.exists(path)
+    assert os.path.exists(os.path.join(settings.LOCAL_REPOSITORY_CACHE, "log.txt"))
+
+
+def _process_bundle(bundle, path, enforce_no_duplicates=True):
+    global logger
     logger = get_logger(bundle)
+
+    # TODO remove
+    logger.setLevel("INFO")
     # get APIs
     salesforce = Salesforce()
     s3 = S3()
-    # download bundle
-    logger.info('Downloading easyDITA bundle from %s', bundle.url)
-    auth = (settings.EASYDITA_USERNAME, settings.EASYDITA_PASSWORD)
-    response = requests.get(bundle.url, auth=auth)
-    zip_file = BytesIO(response.content)
-    unzip(zip_file, path, recursive=True)
+
+    assert os.path.exists(path)
+
+    # get old files representing public state
+    _download_repository_state_from_s3(path)
+
+    # get new files from EasyDITA and put them on top
+    _download_and_unpack_easydita_bundle(bundle, path)
+
     # collect paths to all HTML files
-    html_files = []
-    for dirpath, dirnames, filenames in os.walk(path):
-        for filename in filenames:
-            filename_full = os.path.join(dirpath, filename)
-            if is_html(filename):
-                if skip_html_file(filename):
-                    logger.info(
-                        'Skipping file: %s',
-                        filename_full.replace(path + os.sep, ''),
-                    )
-                    continue
-                html_files.append(filename_full)
-    # check all HTML files and create list of image files
+    html_files = collect_html_paths(path, logger)
+
+    create_drafts(bundle, html_files, path, salesforce, s3)
+
+
+def _find_duplicate_urls(url_map):
+    problems = []
+    if any(map(lambda x: len(x) > 1, url_map.values())):
+        msg = "Found URL name duplicates:"
+        for url_name in sorted(url_map.keys()):
```

Comment:
If you do `for url_name, html_files in sorted(url_map.items()):` then you can use `html_files` instead of `url_map[url_name]` below, which is both more readable and avoids repeating the hash lookup.

Reply:
(I see you're just moving this around, but why not clean it up while it's getting attention.)

Reply:
a890e24caa0bc1200a69d75a73fa997699d40b2c